scipy/optimize/_trustregion_constr/report.py

```diff
@@ -1,10 +1,12 @@
 """Progress report printers."""
 
+from __future__ import annotations
+from typing import List
 
 class ReportBase(object):
-    COLUMN_NAMES = NotImplemented
-    COLUMN_WIDTHS = NotImplemented
-    ITERATION_FORMATS = NotImplemented
+    COLUMN_NAMES: List[str] = NotImplemented
```

Comment:
Hm... the type here definitely doesn't match `NotImplemented`, but I'm also not sure what's up with them being set to `NotImplemented`. Was this supposed to be an ABC or something?

Reply:
It looks to me like it was intended to be an ABC.  Adding types here seems to make mypy do "the right thing", i.e. identify types correctly classes that inherit